treeherder/etl/classification_loader.py

```diff
@@ -0,0 +1,84 @@
+import logging
+import re
+
+import environ
+import newrelic.agent
+
+from treeherder.model.models import MozciClassification, Push, Repository
+from treeherder.utils.taskcluster import download_artifact, get_task_definition
+
+env = environ.Env()
+logger = logging.getLogger(__name__)
+
+CLASSIFICATION_ROUTE_REGEX = re.compile(
+    r"index\.project\.mozci\.classification\.(.+)\.revision\.([0-9A-Fa-f]+)"
+)
+
+
+class ClassificationLoader:
+    def process(self, pulse_job, root_url):
+        task_id = pulse_job["status"]["taskId"]
+
+        task_definition = get_task_definition(root_url, task_id)
+        assert (
+            "routes" in task_definition and len(task_definition["routes"]) > 0
+        ), "A route containing the push project and revision is needed to save the mozci classification"
+        # Retrieving a Push object thanks to the project/revision parsed from the task first route
+        try:
+            push = self.get_push(task_definition["routes"][0])
+        except (ValueError, Repository.DoesNotExist, Push.DoesNotExist):
+            return
```

Comment:
Another thought here is, if the Repository doesn't exist then we probably don't care about it because all that we do care about are added to the Repositories fixture file. But if the Push doesn't exist we'd want to wait and try again because it might be a matter of which message Treeherder ingests first (this or the Push).

Reply:
The error logging is already done in the `ClassificationLoader.get_push()` function so I don't think we need to port it there too but it's your pick, if you want me to log the error two times it's easily feasible.

> But if the Push doesn't exist we'd want to wait and try again because it might be a matter of which message Treeherder ingests first (this or the Push).
Edit: I see further down that it's a retryable task :)

I didn't really thought about that, but we could raise the error in line 30 instead of just returning to stop the ingestion. By raising the error, Celery would retry the ingestion up to 10 times.

> Another thought here is, if the Repository doesn't exist then we probably don't care about it because all that we do care about are added to the Repositories fixture file.

As for this part, in my understanding the unicity of a `Push` is assessed from its `revision` AND its `Repository`. That's why fetching the `Repository` first to then retrieve one of its associated `Push` is necessary.  
We really have to retrieve the `Push` object that was classified because it needs to be linked to the `MozciClassification` object through a `ForeignKey`.

Reply:
> The error logging is already done in the `ClassificationLoader.get_push()` function so I don't think we need to port it there too but it's your pick, if you want me to log the error two times it's easily feasible.
> 

You don't need to log it twice but there is a lot of logging in the backend which makes it hard to sift though the logs so I think the more details the better (maybe at least include a task id)?

> > But if the Push doesn't exist we'd want to wait and try again because it might be a matter of which message Treeherder ingests first (this or the Push).
> > Edit: I see further down that it's a retryable task :)
> 
> I didn't really thought about that, but we could raise the error in line 30 instead of just returning to stop the ingestion. By raising the error, Celery would retry the ingestion up to 10 times.
> 
> > Another thought here is, if the Repository doesn't exist then we probably don't care about it because all that we do care about are added to the Repositories fixture file.
> 
> As for this part, in my understanding the unicity of a `Push` is assessed from its `revision` AND its `Repository`. That's why fetching the `Repository` first to then retrieve one of its associated `Push` is necessary. We really have to retrieve the `Push` object that was classified because it needs to be linked to the `MozciClassification` object through a `ForeignKey`.

Yes, that is all true. I think we should make it a retryable task unless the Repository doesn't exist. If it doesn't exist, we don't care about it.

Reply:
Yes, I agree, thanks a lot for your feedback, I'll make the requested changes and push them along with some unit tests asap.